components/resolved_ts/src/advance.rs

```diff
@@ -149,6 +149,7 @@ pub struct LeadershipResolver {
     region_map: HashMap<u64, Vec<Peer>>,
     // region_id -> peers id, record the responses.
     resp_map: HashMap<u64, Vec<u64>>,
+    checking_regions: HashSet<u64>,
```

Comment:
There's no need to make checking_regions as struct's field, because its usage is within the resolve function. 
It's more clear and sometimes efficient to keep it inside the resolve function.